lib/puppet/provider/x509_request/openssl.rb

```diff
@@ -38,21 +38,17 @@ def exists?
   end
 
   def create
-    cmd_args = [
+    options = [
       'req', '-new',
       '-key', resource[:private_key],
       '-config', resource[:template],
       '-out', resource[:path]
     ]
 
-    if resource[:password]
-      cmd_args.push('-passin')
-      cmd_args.push("pass:#{resource[:password]}")
-    end
-
-    cmd_args.push('-nodes') unless resource[:encrypted]
+    options << ['-passin', "pass:#{resource[:password]}"] if resource[:password]
+    options << ['-nodes'] unless resource[:encrypted]
 
-    openssl(*cmd_args)
+    openssl options
```

Comment:
This passed each item of the array as a parameter and now pass a single parameter: the array itself.

Reply:
Yes, that is exactly the point of this pull request :) in essence: if you compare the x509_cert provider and the x509_request provider, you notice they use different forms of passing arguments to the openssl command. This pull request aims to make this consistent between the two providers. This also results in the other change for passing some arguments as tuples.